src/components/buttons/primary-action-button/primary-action-button.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import BaseButtonWrapper, { BaseButtonContent } from '../base-button-wrapper';
+import filterDataAttributes from '../../../utils/filter-data-attributes';
+import styles from './primary-action-button.mod.css';
+
+const getStateClassName = ({ isToggled, isDisabled }) => {
+  if (isDisabled) return styles.disabled;
+  if (isToggled) return styles.active;
+  return styles.button;
+};
+
+const getSizeClassName = size => styles[size];
+
+const PrimaryActionButton = props => {
+  const dataProps = {
+    'data-track-component': 'PrimaryActionButton',
+    ...filterDataAttributes(props),
+  };
+
+  return (
+    <BaseButtonWrapper
+      id={props.id}
+      name={props.name}
+      ref={props.ref}
+      ariaLabel={props.ariaLabel}
+      onClick={props.onClick}
+      dataAttr={{
+        ...dataProps,
+        ...props.dataAttr,
+      }}
+      isDisabled={props.isDisabled}
+      isToggled={props.isToggled}
+      type={props.type}
+    >
+      <BaseButtonContent
+        styles={classnames(
```

Comment:
should we use `className`?

Reply:
I wanted to differentiate from `className`since this is being passed down to the component